### ISSUE TITLE
Changing the font-size or font-weight on a button elements does nothing

### ISSUE THREAD

**Issue by igamer609** [face-hh/webx#97]
So im trying to make a little header for my site (to put on all my pages)
![image](https://github.com/face-hh/webx/assets/75636007/5589d082-d64f-4b3d-b6de-69d5d345f289)

(sorry in advanced for weird code formatting idk how to fix it)

this is the html:

> <div class="header">
>     <h1 class="title">ITeam</h1>
>     <div></div>
>     <h4 id="members_link">Members</h4>
>     <h4 id="contact_link">Contact</h4>
>     
>     <button id="order_button">Order a product</button>
>   </div>

(ignore the other stuff, only the button matters)

and css:

> header {
>     direction: row;
>     align-items: center;
> 
>     gap: 30;
> 
>     background-color: #111111;
> 
>     padding: 30px;
> 
>     border-radius: 10px;
> }
> 
> button {
>     margin: 20px;
>     width: 70px;
>     height: 40px;
> 
>     font-size: 14px;
>     font-weight: 600;
> }

(yes i tried to use width and height on buttons but its just expermental it does nothing)

Well, as you can see, the text on the button DOES NOT scale up, and also the weight stays the same.
Am i doing something wrong in the code? Is this just not supported? If not, then **why**
 
Fix this, or if there is a fix, **PLS TELL ME!!** i beg you

ok thanks in advanced!!

**Comment by Not-Alaksa-Airlines** [comment]
While I can't see anything wrong, code syntax might be wrong, so try double-checking that.

**Comment by igamer609** [comment]
> While I can't see anything wrong, code syntax might be wrong, so try double-checking that.

no... whatever i try, it doesn't work. I think its just impossible to resize the text on a button lmao

**Comment by Not-Alaksa-Airlines** [comment]
> > While I can't see anything wrong, code syntax might be wrong, so try double-checking that.
> 
> no... whatever i try, it doesn't work. I think its just impossible to resize the text on a button lmao

Maybe you could try formatting a link (the a element). I'm not sure.

**Comment by GStudiosX2** [comment]
Use class instead of ids

**Comment by igamer609** [comment]
> Use class instead of ids. IDs aren't supported

... the id is not in use, the CSS tag is on all buttons on the page. It doesn't work. Font size and wight just don't work for me for some reason (maybe because i compiled it myself something happened)

**Comment by GStudiosX2** [comment]
> > Use class instead of ids. IDs aren't supported
> 
> ... the id is not in use, the CSS tag is on all buttons on the page. It doesn't work. Font size and wight just don't work for me for some reason (maybe because i compiled it myself something happened)

Did you try using a class instead I think it works for me

**Comment by igamer609** [comment]
> > > Use class instead of ids. IDs aren't supported
> > 
> > 
> > ... the id is not in use, the CSS tag is on all buttons on the page. It doesn't work. Font size and wight just don't work for me for some reason (maybe because i compiled it myself something happened)
> 
> Did you try using a class instead I think it works for me

hm alright ill try it, if it works i will do some unspeakable things

**Comment by i-am-JJ** [comment]
The issue you're experiencing is likely due to a missing unit in your CSS. In your CSS, you've set the gap property to 30, but you need to specify a unit such as px (pixels) or em (relative to the font-size of the element).

Regarding the button text not scaling up, it's because you've set a fixed font-size and width for the button. If you want the text to scale up when the button size changes, you should remove the fixed font-size and width properties from the button style.

Here's the corrected CSS:

```css
header {
  display: flex; /* Use flex instead of direction: row; */
  align-items: center;
  gap: 30px; /* Added px unit */
  background-color: #111111;
  padding: 30px;
  border-radius: 10px;
}

button {
  margin: 20px;
  /* Removed fixed width and height */
  font-weight: 600;
}
```

With these changes, the button text should scale up when the button size changes, and the gap between header elements will be 30 pixels.

**Comment by igamer609** [comment]
Well, if you assign the button a custom tag 
`<button class="custom_button">example</button>`
and then you change the parameters in the css for `custom_button {}`, it works!
But button tag needs some love cause it's in depression rn
